src/validation.cpp

```diff
@@ -1891,7 +1891,7 @@ static int64_t nTimeConnect = 0;
 static int64_t nTimeIndex = 0;
 static int64_t nTimeCallbacks = 0;
 static int64_t nTimeTotal = 0;
-static int64_t nBlocksTotal = 0;
+static int64_t nBlocksTotal = 1;
```

Comment:
nit: How about instead introducing a method that returns e.g. the `PerBlockTime(nTimeChainState)`, and handles the 0 case? More lines but the logged numbers remain accurate.

Reply:
In a previous version solved this using a slightly larger change, but I received the following feedback from @gmaxwell:

> This change would have been better done-- easier to review for correctness, more stable against "regression"-- by simply changing the existing initialization constant for nBlocksTotal from 0 to 1.

I think that is a good point. Since this is `validation.cpp` I'd like to keep the changes to the very minimum :)

Reply:
I don't totally buy this as removing the ubsan exception helps protect against regression / div-by-zero. Granted it's more lines and so more review in theory, but only marginally so as you have to look over all the `nBlocksTotal` uses anyway to be clear about the implications, and in both cases you can see only log lines are affected, etc. But yes this is another subjective point.
https://github.com/bitcoin/bitcoin/compare/master...Empact:mills-per-block